src/middlewares/verifyToken.js

```diff
@@ -0,0 +1,80 @@
+import JWT from 'jsonwebtoken';
+import 'dotenv/config';
+import { errorResponse } from '../utils';
+import models from '../models';
+
+/**
+ * @class Authenticate
+ * @description authenticate tokens and roles
+ * @exports Authenticate
+ */
+class Authenticate {
+  /**
+   * Verify if token is valid
+   * @param  {object} req - The user request object
+   * @param  {object} res - The user res response object
+   * @param  {function} next - The next() Function
+   * @returns {String} req.userId - The user id
+   */
+  /* Uncomment this block to verify tokens
+  static verifyToken(req, res, next) {
+    const { headers: { authorization } } = req;
+    const token = authorization.split(' ')[1];
+    try {
+      const decoded = JWT.verify(token, process.env.SECRET);
+      req.userId = decoded;
+      return next();
+    } catch (error) {
+      return errorResponse(res, 500, 'We could not verify your identity. Please try again later');
+    }
+  }
+  */
+  /**
+   * Verify if role is Travel Admin
+   * @param  {object} req - The user request object
+   * @param  {object} res - The user res response object
+   * @param  {function} next - The next() Function
+   * @returns {String} req.userId - The user id
+   */
+  static async verifyTravelAdmin(req, res, next) {
+    const { headers: { authorization } } = req;
+    const token = authorization.split(' ')[1];
+    try {
+      const decoded = JWT.verify(token, process.env.SECRET);
+      const { userId } = decoded;
+      const verify = await models.Profiles.findOne({
+        where: { userId, role: 'Travel Admin' },
+      });
+      if (!verify) {
+        return errorResponse(res, 401, 'Access denied.');
+      }
+      return next();
+    } catch (error) {
+      return errorResponse(res, 500, 'We could not verify your identity. Please try again later');
+    }
+  }
+}
+
+export default Authenticate;
+
+
+// export default async (req, res, next) => {
+//   try {
+//     const { headers: { authorization } } = req;
+//     const token = authorization.split(' ')[1];
+//     if (!token || token === '') {
+//       return errorResponse(res, 401, 'Access denied.');
+//     }
+//     const { userId } = await JWT.verify(token, process.env.SECRET);
+//     const checkAdmin = await models.Profiles.findOne({
+//       where: { userId, role: 'Travel Admin' },
+//     });
+//     if (!checkAdmin) {
+//       return errorResponse(res, 401, 'Access denied.');
+//     }
+//     req.userId = userId;
+//     return next();
+//   } catch (error) {
+//     return errorResponse(res, 500, 'We could not verify your identity. Please try again later');
+//   }
+// };
```

Comment:
welldone @sa-ma your PR works as described, but can you remove the commented codes if they are not needed anymore

Reply:
It would be used by anyone that needs to verify a token. The reason it is commented out is that there are currently no endpoints to utilize it which means it will affect the test coverage. I can take it out if it is an issue

Reply:
ok @sa-ma...

Reply:
You didn't approve :cry: